File: JavaScript/0212.js
function mySqrt(x) {
    let left = 0;
    let right = x;
  
    while (left <= right) {
      const mid = Math.floor((left + right) / 2);
      const midSquared = mid * mid;
  
      if (midSquared === x) {
        return mid;
      } else if (midSquared < x) {
        left = mid + 1;
      } else {
        right = mid - 1;
      }
    }
  
    return right;
  }
  
  console.log(mySqrt(4));  // 輸出: 2
  console.log(mySqrt(8));  // 輸出: 2

File: Python/0212.py
class Solution:
    def mySqrt(self, x: int) -> int:
        left, right = 0, x

        while left <= right:
            mid = (left + right) // 2
            mid_squared = mid * mid

            if mid_squared == x:
                return mid
            elif mid_squared < x:
                left = mid + 1
            else:
                right = mid - 1

        return right

# 使用範例
solution = Solution()
print(solution.mySqrt(4))  # 輸出: 2
print(solution.mySqrt(8))  # 輸出: 2